src/core/server/systems/job.ts

```diff
@@ -423,6 +423,13 @@ export class Job {
             }
         }
 
+        // Check if is engine on in vehicle
+        if (isFlagEnabled(objective.criteria, JobEnums.ObjectiveCriteria.VEHICLE_ENGINE_OFF)) {
+            if (this.player && this.player.vehicle.engineOn) {
```

Comment:
there's an edge case here where this might not hit the `vehicle` property. Need to check if vehicle is defined as well.

It should also be inverted.

```ts
if (!this.player || !this.player.vehicle || this.player.vehicle.engineOn) {
     return false;
}
```